cosmwasm/enclaves/shared/cosmwasm-types/v0.10/src/types.rs

```diff
@@ -119,6 +119,7 @@ pub struct TransactionInfo {
 pub struct BlockInfo {
     pub height: u64,
     /// Absolute time of the block creation in seconds since the UNIX epoch (00:00:00 on 1970-01-01 UTC).
+    /// Assaf: Since we added CosmWasm v1 this is in NANOseconds
```

Comment:
Is it nanoseconds across the board now, or only in v1 contracts?

Reply:
The base type comes in nanos, then for v010 it's converted to seconds and v1 gets nanos + seconds

Reply:
In Tendermint it was always nanos, but in the beginning CosmWasm decided to pass it to the contract as seconds. Somewhere after v0.10 they decided to just pass it as is from Tendermint to the contract.

I added the comment here because on our implementation, the Go side passes everything to the enclave like it's a v0.10 contract (using the v0.10 types), and in the enclave if we detect that it's a v1 contract we convert the Env to its v1 types. When we added v1 we changed it so the Go side passes this field as nanos instead of seconds (we convert to seconds before passing it to a v0.10 contract), but the code comments inside the enclave still said it's in seconds.

Reply:
Having both comments like that is pretty confusing 😅

Reply:
Will be fixed, thanks!